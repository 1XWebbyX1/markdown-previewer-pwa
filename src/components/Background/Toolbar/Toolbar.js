import React from 'react'
import $ from 'jquery'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faExpandArrowsAlt} from '@fortawesome/free-solid-svg-icons'
//import '@fortawesome/fontawesome-free/css/all.css'

class Toolbar extends React.Component {

  render() {
    return (
       <div class="toolbar">
          <h3>{this.props.text}</h3>
          <div onClick={this.props.onClick}><FontAwesomeIcon className='arrows' icon={faExpandArrowsAlt} /></div>
       </div>
    );
  }
}


export default Toolbar;
