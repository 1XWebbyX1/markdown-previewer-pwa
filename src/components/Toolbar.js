import React from 'react'
import $ from 'jquery'
import '@fortawesome/fontawesome-free/css/all.css'

class Toolbar extends React.Component {

  render() {
    return (
       <div class="toolbar">
          <h3>{this.props.text}</h3>
          <i className='arrows fa fa-expand-arrows-alt' onClick={this.props.onClick}></i>
       </div>
    );
  }
}


export default Toolbar;
