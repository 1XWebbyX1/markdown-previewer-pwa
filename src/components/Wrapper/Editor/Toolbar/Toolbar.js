import React from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faInfoCircle, faArrowLeft, faBold, faItalic, faLink, faCode, faImage, faToggleOn} from '@fortawesome/free-solid-svg-icons'



class Toolbar extends React.Component{
  constructor(props){
    super(props);
    this.icon = faInfoCircle;
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    this.icon = (this.icon === faInfoCircle) ? faArrowLeft : faInfoCircle;
    this.props.toggleInfo();
  }

  render(){
    return(
     <div className={'toolbar ' + this.props.className} >
        <ul>
         <li><FontAwesomeIcon className='icon info' icon={this.icon} onClick={this.handleClick}/></li>
         <li><FontAwesomeIcon className='icon bold' icon={faBold} onClick={this.props.onClick}/></li>
         <li><FontAwesomeIcon className='icon italic' icon={faItalic} onClick={this.props.onClick}/></li>
         <li><FontAwesomeIcon className='icon img' icon={faImage} onClick={this.props.onClick}/></li>
         <li><FontAwesomeIcon className='icon link' icon={faLink} onClick={this.props.onClick}/></li>
         <li><FontAwesomeIcon className='icon code' icon={faCode} onClick={this.props.onClick}/></li>
         <li><FontAwesomeIcon className='icon toggle' icon={faToggleOn} onClick={this.props.switchTheme}/></li>
       </ul>
      </div>

    )
  }
}

export default Toolbar;
