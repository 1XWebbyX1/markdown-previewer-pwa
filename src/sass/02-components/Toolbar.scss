.toolbar {
  background: var(--toolbar);
  color: var(--textFont);
  box-shadow: inset 2px 5px 20px #2e2e2e;
  margin-top: -5%;
  height: 6.5vmax;

  .arrows{
    display: none;
  }

  h3 {
    user-select: none;
    padding: 4%;
    padding-right: 0;
    font-size: 2.4vmax;
    text-shadow: 0 -1px 0 #fff, 0 1px 0 #2e2e2e, 0 2px 0 #2c2c2c,
      0 3px 0 #2a2a2a, 0 4px 0 #282828;
  }
}
