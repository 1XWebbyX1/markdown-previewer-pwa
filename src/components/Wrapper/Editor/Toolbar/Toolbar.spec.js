// External Depedencies
import React from 'react';
import { shallow, configure } from 'enzyme';
// Our Dependencies
import { expect } from '../../utils/chai';
import Adapter from 'enzyme-adapter-react-16';
import renderer from "react-test-renderer";

// Our Component
import Toolbar from './Toolbar';


configure({ adapter: new Adapter() });

describe('Toolbar', () => {

  it("renders correctly", () => {
     const output = renderer.create(
         <Toolbar />
   ).toJSON();
     expect(output).to.matchSnapshot();
  });

  it("renders 7 list elements", () => {
    const app = shallow(<Toolbar />);
    expect(app).to.have.exactly(7).descendants('li');
  })
})
