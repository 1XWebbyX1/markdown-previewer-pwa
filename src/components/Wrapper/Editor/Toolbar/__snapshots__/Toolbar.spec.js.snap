// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar renders correctly 1`] = `
<div
  className="toolbar undefined"
>
  <ul>
    <li>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-info-circle fa-w-16 icon info"
        data-icon="info-circle"
        data-prefix="fas"
        onClick={[Function]}
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </li>
    <li>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-bold fa-w-12 icon bold"
        data-icon="bold"
        data-prefix="fas"
        role="img"
        style={Object {}}
        viewBox="0 0 384 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M304.793 243.891c33.639-18.537 53.657-54.16 53.657-95.693 0-48.236-26.25-87.626-68.626-104.179C265.138 34.01 240.849 32 209.661 32H24c-8.837 0-16 7.163-16 16v33.049c0 8.837 7.163 16 16 16h33.113v318.53H24c-8.837 0-16 7.163-16 16V464c0 8.837 7.163 16 16 16h195.69c24.203 0 44.834-1.289 66.866-7.584C337.52 457.193 376 410.647 376 350.014c0-52.168-26.573-91.684-71.207-106.123zM142.217 100.809h67.444c16.294 0 27.536 2.019 37.525 6.717 15.828 8.479 24.906 26.502 24.906 49.446 0 35.029-20.32 56.79-53.029 56.79h-76.846V100.809zm112.642 305.475c-10.14 4.056-22.677 4.907-31.409 4.907h-81.233V281.943h84.367c39.645 0 63.057 25.38 63.057 63.057.001 28.425-13.66 52.483-34.782 61.284z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </li>
    <li>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-italic fa-w-10 icon italic"
        data-icon="italic"
        data-prefix="fas"
        role="img"
        style={Object {}}
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M204.758 416h-33.849l62.092-320h40.725a16 16 0 0 0 15.704-12.937l6.242-32C297.599 41.184 290.034 32 279.968 32H120.235a16 16 0 0 0-15.704 12.937l-6.242 32C96.362 86.816 103.927 96 113.993 96h33.846l-62.09 320H46.278a16 16 0 0 0-15.704 12.935l-6.245 32C22.402 470.815 29.967 480 40.034 480h158.479a16 16 0 0 0 15.704-12.935l6.245-32c1.927-9.88-5.638-19.065-15.704-19.065z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </li>
    <li>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-image fa-w-16 icon img"
        data-icon="image"
        data-prefix="fas"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M464 448H48c-26.51 0-48-21.49-48-48V112c0-26.51 21.49-48 48-48h416c26.51 0 48 21.49 48 48v288c0 26.51-21.49 48-48 48zM112 120c-30.928 0-56 25.072-56 56s25.072 56 56 56 56-25.072 56-56-25.072-56-56-56zM64 384h384V272l-87.515-87.515c-4.686-4.686-12.284-4.686-16.971 0L208 320l-55.515-55.515c-4.686-4.686-12.284-4.686-16.971 0L64 336v48z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </li>
    <li>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-link fa-w-16 icon link"
        data-icon="link"
        data-prefix="fas"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </li>
    <li>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-code fa-w-20 icon code"
        data-icon="code"
        data-prefix="fas"
        role="img"
        style={Object {}}
        viewBox="0 0 640 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M278.9 511.5l-61-17.7c-6.4-1.8-10-8.5-8.2-14.9L346.2 8.7c1.8-6.4 8.5-10 14.9-8.2l61 17.7c6.4 1.8 10 8.5 8.2 14.9L293.8 503.3c-1.9 6.4-8.5 10.1-14.9 8.2zm-114-112.2l43.5-46.4c4.6-4.9 4.3-12.7-.8-17.2L117 256l90.6-79.7c5.1-4.5 5.5-12.3.8-17.2l-43.5-46.4c-4.5-4.8-12.1-5.1-17-.5L3.8 247.2c-5.1 4.7-5.1 12.8 0 17.5l144.1 135.1c4.9 4.6 12.5 4.4 17-.5zm327.2.6l144.1-135.1c5.1-4.7 5.1-12.8 0-17.5L492.1 112.1c-4.8-4.5-12.4-4.3-17 .5L431.6 159c-4.6 4.9-4.3 12.7.8 17.2L523 256l-90.6 79.7c-5.1 4.5-5.5 12.3-.8 17.2l43.5 46.4c4.5 4.9 12.1 5.1 17 .6z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </li>
    <li>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-toggle-on fa-w-18 icon toggle"
        data-icon="toggle-on"
        data-prefix="fas"
        role="img"
        style={Object {}}
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M576 256c0 106.039-85.961 192-192 192H192C85.961 448 0 362.039 0 256S85.961 64 192 64h192c106.039 0 192 85.961 192 192zM384 128c-70.741 0-128 57.249-128 128 0 70.741 57.249 128 128 128 70.741 0 128-57.249 128-128 0-70.741-57.249-128-128-128"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </li>
  </ul>
</div>
`;
